### ISSUE TITLE
Received images not coming through

### ISSUE THREAD

**Issue by jessejoe** [yakyak/yakyak#407]
Someone sent me an image, which I can see on Hangouts:
![image](https://cloud.githubusercontent.com/assets/1694586/19085722/59f5fece-8a39-11e6-96f8-f3373928a832.png)

Source:

```
<div class="et9sxc" style="width: 164px; height: 218px;"><a href="https://plus.google.com/u/0/photos/albums/p4nndcff37cbtsftc604se2h6em2of3ne?pid=6337664088204164434&amp;oid=103774248546935218356" target="_blank" class="WLrLpb" tabindex="-1"><img src="https://lh3.googleusercontent.com/-13r9S0WY72Q/V_Pn6OsovVI/AAAAAAAAP5o/dh9u5WfzWUAV7ozK0azhJx5Vshs-Rd02wCK8B/s512/IMG_2834.JPG" class="uqvYjb KgFPz" alt="" aria-label="Picture. Press Enter to open it in a new page."></a><div class="glTU3d utCB7e NjzSJc uq"></div></div>
```

But it didn't come through yakyak:
![image](https://cloud.githubusercontent.com/assets/1694586/19085735/65b6f5ba-8a39-11e6-9610-a7367bfd64ba.png)

Looking at the source, I don't see an image/URL:

```
<div class="ugroup"><a href="https://plus.google.com/u/0/12345/about" title="Michael Smith" class="sender"><div class="initials">MS</div></a><div class="umessages"><div id="7-H0Z7-AG5j8Jc4LOBkGeS" title="Today at 1:33 PM" class="message chat_message"></div></div></div>
```

I'm using `v1.3.3`.

**Comment by sethpjohnson** [comment]
Having the same problem. Version 1.2.5 (1.3.2)

**Comment by averissimo** [comment]
I guess extra information will be required:

If you guys open the inspector (pressing Ctrl+Shift+I, or via menu YakYak -> Open Inspector)

**Are there any errors shown in the console tab?**

In the console there should also be a message named "attachment" that can be expanded such as:

![image](https://cloud.githubusercontent.com/assets/211358/19243911/b7b29072-8f11-11e6-820a-698b64abe23b.png)

Try to find the one that matches the missing image (if there is one) and **create a screenshot as the one I've done** (censoring the information)

And **test both links to make sure they work.**

tip: when opening the conversation, the last "attachment" should be the latest image in the conversation

**Comment by chasetb** [comment]
This is all I see when I found an image that didn't load in YakYak. It's supposed to be an animated GIF that shows fine in Hangouts. Also using 1.2.5 (1.3.2).
![screen shot 2016-10-10 at 11 01 42 am](https://cloud.githubusercontent.com/assets/1828637/19244303/f5823aee-8ed8-11e6-856e-a1a6be99b471.png)

**Comment by sethpjohnson** [comment]
I noticed that some came through too. I think the ones that were sent from a cellphone (camera) were dropped but a small screen capture came through. Maybe its a size issue?

**Comment by chasetb** [comment]
The ones that aren't coming through for me are typically screenshots or images from the web.

**Comment by averissimo** [comment]
That object usually is more complex *(as my screenshot)* and has at least 2 URLs

1. a thumbnail (that is shown in YakYak) 
1. the full size that you can see once you click on it.

having an empty object could indicate that it is a new type of message that is still not supported in the current api (yakyak/hangupsjs)

I've tested sending a gif (via google's hangouts) between my account and a test one and it actually shows on YakYak

![outce37zqflcn](https://cloud.githubusercontent.com/assets/211358/19245658/30993006-8f1a-11e6-8fa8-1af8d40462d8.gif)

Can you describe a way to replicate this problem and share the image where that happens?

**Comment by jessecarbon** [comment]
Ok, I figured out how to reproduce it. For my case, my friend is sending me messages via the Hangouts messaging window from Gmail. The same thing happens when I share an image from a test account to myself:

![image](https://cloud.githubusercontent.com/assets/20824912/19274773/4416ee04-8f9f-11e6-93b1-6e979a8623ba.png)

Again the message is empty in Hangouts:

![image](https://cloud.githubusercontent.com/assets/20824912/19274809/6a01fc30-8f9f-11e6-98c5-a90b0de31937.png)

I have many attachments in my console, but none of them seem to contain a URL like when sharing via the Hangouts mobile app itself.

So it's easily reproducible @averissimo, just share an image with the Gmail Hangouts interface.

**Comment by averissimo** [comment]
confirm it! :) nice @jessecarbon

**Comment by jessejoe** [comment]
Just wanted to mention that it's not only images sent from the web interface. I just sent a message from my phone to someone via Hangouts and while they received it, it shows empty in yakyak. Also sometimes people send me images from their phones through Hangouts and they don't come through either.

**Comment by averissimo** [comment]
this seems to be a yakyak/hangupsjs issue, I have traced it back to the event from the hangupsjs, with the object lacking the image information.

I've opened a new issue there: https://github.com/yakyak/hangupsjs/issues/57

**Comment by aPoCoMiLogin** [comment]
For me images are not showing when someone is sharing image from phone, like from other application with share button. No issue in web/mobile app, it's just not recived in yakyak.

**Comment by kecinzer** [comment]
I have this issue too. I don't recieve any images. I can normaly see it in Hangouts app or on web.

**Comment by retorquere** [comment]
Same here.

**Comment by jelmerf** [comment]
Both in version 1.3.2 and 1.3.3 none of previously received images show up. Having someone send me an image shows nothing in the chat log.

**Comment by retorquere** [comment]
That, and in many (but not all) cases, the user profile picture doesn't show.

**Comment by Mentis** [comment]
I have same issue

**Comment by averissimo** [comment]
There's hope!!

Just finished with `hangoutsjs` api library and some quick code to YakYak. As a result the embed photo data is available in my **development** version of yakyak 

I'll try to create a binary version in the next few days for users to test, that along with other improvements ;)

Screenshots below of without fix (that shows empty bubles, when lucky) and with fix :D

Some images show as they are uploaded directly to conversation.

### Without fix:

![image](https://cloud.githubusercontent.com/assets/211358/20585654/aff364a0-b1f4-11e6-9287-c320249c6ac9.png)

### With new fix:

![image](https://cloud.githubusercontent.com/assets/211358/20585639/995d5f2a-b1f4-11e6-9c05-69c216162002.png)

**Comment by aPoCoMiLogin** [comment]
@averissimo nice! Is there any build with that changes, that we can test it? Thanks!

**Comment by averissimo** [comment]
Yup!! see here #445

**Comment by averissimo** [comment]
@aPoCoMiLogin can you please leave feedback just to make sure it fixes the problem LoL

and if you can also comment on new features it would be awesome! :D

**Comment by dominiczaq** [comment]
Hello,

I've tried the new build on mac and it seems it's not working (not starting
after launching app).
[image: pasted1]
This is understandable as this is beta build, but still not running on mac.
Anyone else having same problem?

czw., 24.11.2016 o 15:14 użytkownik André Veríssimo <
notifications@github.com> napisał:

**Comment by averissimo** [comment]
Oh crap! Can you reupload the image, as it is not showing in github.com

**Comment by dominiczaq** [comment]
Sure, image below.
![image](https://cloud.githubusercontent.com/assets/7099901/20602022/cca86dd6-b25b-11e6-8710-ba34d6dac930.png)

**Comment by aPoCoMiLogin** [comment]
@averissimo it's not working for me too. For example this image in phone:

![hangout in android](http://i.imgur.com/33juFkC.jpg)

![yakyak 1.4.0 beta osx](http://i.imgur.com/BPkGQzP.jpg)

Maybe it's working only for newest images?

**Comment by averissimo** [comment]
LOL sorry guys!!

Merged all pull requests except for this fix! -_-

Deploying new files...

**Comment by averissimo** [comment]
Ok, tested the binaries in linux and windows virtual machine.. I replaced the files in the release, so it should work great in Linux and Windows environments.

I cannot test it in MacOSX though, so I have to ask someone that can build and test it in OSX to help out.

Please download again and test it. (it is expected to work with previous images)

ps. It will have some extra debugging in DevTools Console, but just ignore it as it does not affect functionality. It was commited by mistake :( ... I don't have time to rebuild again right now

**Comment by averissimo** [comment]
@dominiczaq are the menus available? Can you try to open the Developer Tools and see what error appears in the console tab

An alternative way to get this information is to launch YakYak from the command and take a screenshot of the output (I have no idea on how to launch in the command, though.. sorry)

**Comment by jessejoe** [comment]
@averissimo getting the same blank window when opening on OSX

![image](https://cloud.githubusercontent.com/assets/1694586/20603988/1d3b1764-b232-11e6-9866-6fa839635f93.png)

I see this from the console:
```
$ ~/Downloads/YakYak.app/Contents/MacOS/YakYak
Error on YakYak window:
 { columnNo: 18,
  error: {},
  lineNo: 15,
  msg: 'Uncaught TypeError: Cannot read property \'split\' of undefined',
  url: '/Users/jesse/Downloads/YakYak.app/Contents/Resources/app/ui/version.js' }
--- End of error message in YakYak window.
```

Also, would you prefer these comments in #445?

**Comment by codethought** [comment]
@averissimo your new build on your fork is working well for me so far. Thanks for putting in the effort.

**Comment by averissimo** [comment]
@jessejoe move conversation here #423 as error seems related to this

**Comment by jessejoe** [comment]
Confirming with your fix in https://github.com/yakyak/yakyak/pull/423#issuecomment-262807020 that images are coming through on the OSX `v1.4.0-beta1` build.

**Comment by aPoCoMiLogin** [comment]
Yup can confirm that images are coming through on OSX.

**Comment by averissimo** [comment]
included in 1.4.0 release
